src/main/scala/mesosphere/marathon/core/event/impl/callback/SubscribersKeeperActor.scala

```diff
@@ -38,38 +42,46 @@ class SubscribersKeeperActor(val store: EntityStore[EventSubscribers]) extends A
       subscription pipeTo sender()
 
     case GetSubscribers =>
-      val subscription = store.fetch(Subscribers).map(_.getOrElse(EventSubscribers()))(context.dispatcher)
-
+      val subscription = store.get().map(_.getOrElse(EventSubscribers()))(context.dispatcher)
       import context.dispatcher
       subscription pipeTo sender()
   }
 
   protected[this] def add(callbackUrl: String): Future[EventSubscribers] =
-    store.modify(Subscribers) { deserialize =>
-      val existingSubscribers = deserialize()
-      if (existingSubscribers.urls.contains(callbackUrl)) {
-        log.info("Existing callback {} resubscribed.", callbackUrl)
-        existingSubscribers
-      } else EventSubscribers(existingSubscribers.urls + callbackUrl)
-    }
+    lockManager.executeSequentially(LockName) {
+      async {
+        val subscribers = await(store.get()).getOrElse(EventSubscribers())
+        val updated = if (subscribers.urls.contains(callbackUrl)) {
+          log.info("Existing callback {} resubscribed.", callbackUrl)
+          subscribers
+        } else EventSubscribers(subscribers.urls + callbackUrl)
+
+        if (updated != subscribers) {
+          await(store.store(updated))
+        }
+        updated
+      }(ExecutionContext.global)
+    }(ExecutionContext.global) // blocks a thread, don't block the actor.
```

Comment:
Old:
PersistenceStore would let you provide a `change` method that updated the existing state.

New:
PersistenceStore doesn't provide you a method to change the existing state given the old state. So you have to read it first yourself if you want a "go grab what I stored last time, append something to it, then store it". And you probably want only a single writer allowed to do this at a time.